src/incremental-compilation.md

```diff
@@ -43,24 +43,23 @@ There are two key insights here:
     
 ### The try-mark-green algorithm
 
-The core of the incremental compilation is an algorithm called
+At the core of incremental compilation is an algorithm called
 "try-mark-green". It has the job of determining the color of a given
-query Q (which must not yet have been executed). In cases where Q has
+query Q (which must not have yet been executed). In cases where Q has
 red inputs, determining Q's color may involve re-executing Q so that
-we can compare its output; but if all of Q's inputs are green, then we
-can determine that Q must be green without re-executing it or inspect
-its value what-so-ever. In the compiler, this allows us to avoid
-deserializing the result from disk when we don't need it, and -- in
-fact -- enables us to sometimes skip *serializing* the result as well
+we can compare its output, but if all of Q's inputs are green, then we
+can conclude that Q must be green without re-executing it or inspecting
+its value, regardless. In the compiler, this allows us to avoid
```

Comment:
I actually prefer "whatsoever" or "at all"

Reply:
"Whatsoever" is a conjunctive adverb so needs a clause after it, but I can change it to "whatsoever it may be" or "at all", your choice. :-)

Reply:
Ah, I did not know that :)

Maybe "at all" is best.

Reply:
Oh wait, I read it wrong. And it can be used as a normal adverb as well, it just occurred to me. My bad. We'll go with "at all" anyway. :-P